blackbox/handlers/databases/localstorage.py

```diff
@@ -13,30 +12,33 @@ class LocalStorage(BlackboxDatabase):
 
     required_fields = ("path",)
 
+    def __init__(self, **kwargs) -> None:
+        # Gzip deflates only accept compression level ranging from 0 to 9
+        # if the argument is given, we check it is in the allowed range otherwise
+        # RuntimeError will be raised by ZipFile
+        if compression_level := kwargs.get("compression_level"):
+            if compression_level < 0 or compression_level > 9:
+                raise ImproperlyConfigured(
+                    f"Invalid compression level. Must be an integer between 0 and 9, got {compression_level}."
+                )
+
+        super().__init__(**kwargs)
+
     def backup(self) -> Path:
         date = datetime.date.today().strftime("%d_%m_%Y")
 
         path = self.config["path"]
         compression_level = self.config.get("compression_level", 5)
 
-        if compression_level < 0 or compression_level > 9:
-            raise ImproperlyConfigured(
-                f"Invalid compression level. Must be an integer between 0 and 9, got {compression_level}."
-            )
-
         backup_path = Path.home() / f"{self.config['id']}_blackbox_{date}.zip"
 
+        # Store evey file in the archive
+        # We use deflate (Gzip) for compression and the level has already been validated in __init__
         with ZipFile(backup_path, "w", ZIP_DEFLATED, compresslevel=compression_level) as zipfile:
-            directories = deque((Path(path),))
-
-            while not len(directories) == 0:
-                path = directories.pop()
-
-                for file in path.iterdir():
-                    if file.is_file():
-                        zipfile.write(file)
-                    elif file.is_dir():
-                        directories.append(file)
+            for file in path.rglob("*"):
+                if file.is_file():
```

Comment:
this line irks me a bit. why did we name the variable `file` if we have to check to make sure it is a file? that's why my suggestion had `subpath`, but I'm open to other more "zoomed out" alternatives. I just don't think we should name something file and then check if it's a file.